src/dag.ts

```diff
@@ -32,14 +42,54 @@ const arrToObj = (d: any[], attr: string) => d.reduce((r: any, c: any) => {
   return r;
 }, {});
 
-export function iterate<T>(nodes: INode<T>[], edges: IEdge[],
-                        iteratorFn: (node: T, prev: T[], i: number) => void, untilNode?: string) {
+export type IteratorFn<T> = (node: T, prev: T[], i: number) => void;
+
+/**
+ * Traverse a Directed Acyclic Graph via Depth-first search
+ * @arg nodes array of nodes describing the graph
+ * @arg edges array of edges describing the graph
+ * @arg func iterator callback
+ * @arg untilNode iterate until specific node
+ */
+export function iterateDfs<T>(nodes: INode<T>[], edges: IEdge[], func: IteratorFn<T>, untilNode?: string) {
+  const options: IIteratorOptions = {
+    traversalMode: TraversalMode.DFS,
+    untilNode: untilNode,
```

Comment:
this line can be just untilNode, similar in the bfs method

Reply:
cool, I didn't know you can do this. However, I think here using `{key: val}` is more verbose and avoids the error of ordering the properties.

Reply:
sounds good.